lxqtbacklight/linux_backend/driver/libbacklight_backend.c

```diff
@@ -165,7 +179,27 @@ static int read_bl_power(const char *driver)
     return read_int("bl_power", driver);
 }
 
-typedef enum {FIRMWARE, PLATFORM, RAW, OTHER, N_BACKLIGHT} BackligthTypes;
+static int read_type(const char *driver)
+{
+    FILE *in = open_driver_file("type", driver, "r");
+    if( in == NULL ) {
+        return -1;
+    }
+    char type[1024];
```

Comment:
1024 is probably overkill here, but the same value is used in `lxqt_backlight_backend_get_driver()` as well.